### PR TITLE
[macOS] Fix crash handler not printing function names on M1 Macs.

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes incorrect architecture name used to call `atos`.

### CODE REVIEW

File: platform/osx/crash_handler_osx.mm [hunk 1]
```diff
@@ -135,7 +135,11 @@ static void handle_crash(int sig) {
 				args.push_back("-o");
 				args.push_back(_execpath);
 				args.push_back("-arch");
+#if defined(__x86_64) || defined(__x86_64__) || defined(__amd64__)
 				args.push_back("x86_64");
+#elif defined(__aarch64__)
+				args.push_back("arm64");
```
Comment (akien-mga) [inline platform/osx/crash_handler_osx.mm:141]: That shouldn't happen, but in theory of new arches were added then the call would become invalid as there's no `#else` condition. Maybe it should raise an error in `#else`?

Or is it mandatory to set the arch, or can we get away with just letting it infer things?

Comment (bruvzg) [reply platform/osx/crash_handler_osx.mm:142]: Moved `args.push_back("-arch");` into the ifdef, it's an optional flag. This way it should try to infer architecture if we do not know it.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by bruvzg** [comment]
This fix is relevant for 3.x as well.

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.1.

**Pull request merged**